files/wsl-service

```
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# This script is used to manage an (OpenRC) service on WSL.  This is required as
+# we need to put services in separte PID namespaces there, so using rc-service
+# directly will not return correct results.
+
+set -o errexit -o nounset
+
+# `/usr/local/bin/wsl-init` writes the PID of the _parent_ (unshare) process
+# into /run/wsl-init.pid; so we'll need to look at its children that is init
+# and spawn rc-service within that PID namespace in order to talk to OpenRC
+# correctly.
+
+parent="$(cat /run/wsl-init.pid)"
+pid="$(ps -o pid,ppid,comm | awk '$2 == "'"${parent}"'" && $3 == "init" { print $1 }')"
```

Should we throw an error when `pid` is the empty string? Or is `nsenter: invalid number ''` good enough?

---

Changed in b9706f4b5c333f74056fe71c45cfd1eac9eb7063 — I was going to say that this is only run by RD, but even in that case having an easier to understand error message seems like a good idea.